models/turbine_models/custom_models/sdxl_inference/unet.py

```diff
@@ -70,9 +72,101 @@ def forward(
         return noise_pred
 
 
+def get_punet_model(hf_model_name, external_weight_path, precision="i8"):
+    from sharktank.models.punet.model import (
+        Unet2DConditionModel as punet_unet,
+        ClassifierFreeGuidanceUnetModel as CFGPunetModel,
+    )
+
+    if precision == "i8":
+        repo_id = "amd-shark/sdxl-quant-models"
+        subfolder = "unet/int8"
+        revision = "82e06d6ea22ac78102a9aded69e8ddfb9fa4ae37"
+    elif precision in ["fp16", "fp32"]:
+        repo_id = hf_model_name
+        subfolder = "unet"
+        revision = "76d28af79639c28a79fa5c6c6468febd3490a37e"
+
+    def download(filename):
+        return hf_hub_download(
+            repo_id=repo_id, subfolder=subfolder, filename=filename, revision=revision
+        )
+
+    results = {
+        "config.json": download("config.json"),
+        "params.safetensors": download("params.safetensors"),
+    }
+    if precision == "i8":
+        results["quant_params.json"] = download("quant_params.json")
+        output_path = external_weight_path.split("unet")[0] + "punet_dataset_i8.irpa"
+        ds = get_punet_i8_dataset(
+            results["config.json"],
+            results["quant_params.json"],
+            results["params.safetensors"],
+            output_path,
+            base_params=None,
+        )
+    else:
+        ds = None  # get_punet_dataset(results["config.json"], results["params.safetensors"], base_params=None)
+
+    cond_unet = punet_unet.from_dataset(ds)
+    mdl = CFGPunetModel(cond_unet)
+    return mdl
+
+
+def get_punet_i8_dataset(
+    config_json_path,
+    quant_params_path,
+    params_path,
+    output_path="./punet_dataset_i8.irpa",
+    quant_params_struct=None,
+    base_params=None,
+):
+    from sharktank.models.punet.tools.import_brevitas_dataset import (
```

Comment:
Please don't do it this way. Just import the `main` function and pass it CL args vs using the private internals.